### PR TITLE
Feature - Update Support Form error

### LINKED ISSUE #4231: New message for support form error
## Description

If freshdesk request contains an error, show toast to user saying to email us, and mention this error code.

## Copy

- English: "Sorry, something went wrong. Please email {emailAddress} and mention this error code: {errorCode}."
- French: "Désolé. Une erreur s'est produite. Veuillez envoyer un courriel à {emailAddress} et mentionner ce code d’erreur : {errorCode}."

## Acceptance Criteria
- [x] Shows the error code
- [x] Directs user to email our support email ([gctalent-talentgc@support-soutien.gc.ca](mailto:gctalent-talentgc@support-soutien.gc.ca))

### PR DESCRIPTION
**Pull request by mnigh**
Resolves #4231.

## Notes
In addition to updating the support form error message in English and adding its French string translation, this also [adds TALENTSEARCH_SUPPORT_EMAIL environment variable](https://github.com/GCTC-NTGC/gc-digital-talent/commit/71ce5a00a17a6fd15444f3c2912a0bcc6333df82).

## Steps to test

1. Compile talentsearch `npm run intl-compile --workspace=talentsearch`
2. Build talentsearch `npm run production --workspace=talentsearch`
3. Navigate to **/support**
4. Force an error by commenting out required `email` parameter in [SupportController](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/api/app/Http/Controllers/SupportController.php#L15)
5. Submit form 
6. Verify error toast message in English and French

### CODE REVIEW

File: frontend/talentsearch/src/js/components/support/SupportForm.tsx [hunk 1]
```diff
@@ -235,11 +238,14 @@ const SupportFormApi = () => {
...
-            id: "C3Lv2t",
+            defaultMessage: `Sorry, something went wrong. Please email {emailAddress} and mention this error code: {errorCode}.`,
+            id: "tZLItl",
             description: "Support form toast message error",
           },
-          { errorCode: response.status },
+          {
+            emailAddress: TALENTSEARCH_SUPPORT_EMAIL,
+            errorCode: response.status,
+          },
```
Comment (esizer) [inline frontend/talentsearch/src/js/components/support/SupportForm.tsx:248]: Can we add a `mailto:` link to the email address and extend how long the toast appears for or force it to be dismissed?

I had a hard time remembering the email address in the short amount of time it appeared for and trying to copy the email just dismissed the toast for me.

**Review by esizer — CHANGES_REQUESTED** [review]
Looking good, just a little difficult to use the email address presented.

File: frontend/talentsearch/src/js/components/support/SupportForm.tsx (hunk 1 above)
Comment (mnigh) [reply frontend/talentsearch/src/js/components/support/SupportForm.tsx:248]: Yep, the [toast](https://fkhadra.github.io/react-toastify/api/toast) api has options including `autoClose` ("Delay in ms to close the toast. If set to false, the notification needs to be closed manually"), how long did you have in mind to dely the close? I don't think we currently have anywhere else on the site where we don't use the parameters set in [ToastContainer](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/frontend/common/src/components/Toast/Toast.tsx#L32).

Comment (esizer) [reply frontend/talentsearch/src/js/components/support/SupportForm.tsx:248]: Rule of thumb I was taught was enough time to read the message 3 times over. Took me about 7 seconds to read so 20 seconds should be enough. We can try it out and see if that is too long or not.

Comment (mnigh) [reply frontend/talentsearch/src/js/components/support/SupportForm.tsx]: See latest changes:

- [Add anchorTag to support form toast message error](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4450/commits/1ae68c83cc6664fade3052b16fd3042e7d1facfb)
- [Delay toast error autoClose for 20 seconds](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4450/commits/7ba5b71eba76fa69ddf416a73af060ead8ade17d)

**Review by esizer — APPROVED** [review]
Sweet 🍭

**Pull request merged**